server/src/main/java/org/opensearch/search/aggregations/DefaultAggregationProcessor.java

```diff
@@ -6,112 +6,46 @@
  * compatible open source license.
  */
 
-/*
- * Licensed to Elasticsearch under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-/*
- * Modifications Copyright OpenSearch Contributors. See
- * GitHub history for details.
- */
-
 package org.opensearch.search.aggregations;
 
 import org.apache.lucene.search.Collector;
-import org.apache.lucene.search.CollectorManager;
 import org.apache.lucene.search.Query;
-import org.opensearch.common.inject.Inject;
 import org.opensearch.common.lucene.search.Queries;
-import org.opensearch.search.aggregations.bucket.global.GlobalAggregator;
 import org.opensearch.search.internal.SearchContext;
-import org.opensearch.search.profile.aggregation.ProfilingAggregator;
 import org.opensearch.search.profile.query.CollectorResult;
 import org.opensearch.search.profile.query.InternalProfileCollector;
 import org.opensearch.search.query.QueryPhaseExecutionException;
-import org.opensearch.search.query.ReduceableSearchResult;
 
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 
 /**
- * Aggregation phase of a search request, used to collect aggregations
- *
- * @opensearch.internal
+ * {@link AggregationProcessor} implementation which is used with {@link org.opensearch.search.query.QueryPhase.DefaultQueryPhaseSearcher}.
+ * This is the default implementation which works when collection for aggregations happen in sequential manner. It doesn't perform any
+ * reduce on the collected documents at shard level
  */
-public class AggregationPhase {
-
-    @Inject
-    public AggregationPhase() {}
-
+public class DefaultAggregationProcessor implements AggregationProcessor {
```

Comment:
I believe we are breaking the semantics of the search phases here, the `AggregationPhase` name was approriate

Reply:
It does not look like separated phase indeed but it looks consistent with other steps (phases?) the query phase was doing: RescorePhase, SuggestPhase, ... (and before AggregationPhase).

Reply:
I am keeping this as is because Suggest, Rescore and Aggregations are actors for the `QueryPhase`. The phases for a search request are `Query` or `Fetch`. I think we should consider renaming these other actors to processors as well

Reply:
:+1: I am all for consistency, if you think renaming better reflects the purpose of this classes, let's do that separately for all impacted ones, right now it brings inconsistency.

Reply:
Created [issue](https://github.com/opensearch-project/OpenSearch/issues/7854) to track this.